src/main/java/org/prebid/server/settings/SettingsCache.java

```diff
@@ -20,20 +22,25 @@ public class SettingsCache implements CacheNotificationListener {
     private final Map<String, Set<StoredItem>> requestCache;
     private final Map<String, Set<StoredItem>> impCache;
 
-    public SettingsCache(int ttl, int size) {
+    public SettingsCache(int ttl, int size, int jitter) {
         if (ttl <= 0 || size <= 0) {
             throw new IllegalArgumentException("ttl and size must be positive");
         }
-        requestCache = createCache(ttl, size);
-        impCache = createCache(ttl, size);
+        requestCache = createCache(ttl, size, jitter);
+        impCache = createCache(ttl, size, jitter);
     }
 
-    public static <T> Map<String, T> createCache(int ttl, int size) {
-        return Caffeine.newBuilder()
+    public static <T> Map<String, T> createCache(int ttl, int size, int jitter) {
+        final Cache<String, T> cache = Caffeine.newBuilder()
                 .expireAfterWrite(ttl, TimeUnit.SECONDS)
                 .maximumSize(size)
-                .<String, T>build()
-                .asMap();
+                .build();
+        cache.policy()
+                .expireAfterWrite()
+                .ifPresent(expiration -> expiration.setExpiresAfter(
+                        jitter != 0 ? ttl + ThreadLocalRandom.current().nextLong(jitter * -1, jitter) : ttl,
+                        TimeUnit.SECONDS));
```

Comment:
This definitely should be before build statement

Reply:
Unfortunately it is impossible. I do not have access to policy setup before cache instance build.